### ISSUE TITLE
`textwidth` is deprecated warning

### ISSUE THREAD

**Issue by ExpandingMan** [JuliaData/DataFrames.jl#1324]
I get the following warning when displaying a DataFrame with strings
```
WARNING: textwidth is deprecated, use Compat.Unicode.textwidth instead.    
Stacktrace:                      
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:70 
 [2] textwidth(::String, ::Vararg{String,N} where N) at ./deprecated.jl:31
 [3] getmaxwidths(::DataFrames.DataFrame, ::UnitRange{Int64}, ::UnitRange{Int64}, ::Symbol) at /home/expandingman/.julia/v0.6/DataFrames/src/abstractdataframe/show.jl:96
 [4] show(::IOContext{Base.Terminals.TTYTerminal}, ::DataFrames.DataFrame, ::Bool, ::Symbol, ::Bool) at /home/expandingman/.julia/v0.6/DataFrames/src/abstractdataframe/show.jl:450
 [5] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::DataFrames.DataFrame) at ./REPL.jl:122
 [6] display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::DataFrames.DataFrame) at ./REPL.jl:125
 [7] display(::DataFrames.DataFrame) at ./multimedia.jl:218               
 [8] eval(::Module, ::Any) at ./boot.jl:235                                                                                                                                                                           [9] print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:144
 [10] print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:129                                                                                                                        [11] (::Base.REPL.#do_respond#16{Bool,Base.REPL.##26#36{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:646                                                                                                                                             
 [12] eval(::Module, ::Any) at ./boot.jl:235                                                                                        
 [13] run_interface(::Base.Terminals.TTYTerminal, ::Base.LineEdit.ModalInterface) at ./LineEdit.jl:1583
 [14] run_frontend(::Base.REPL.LineEditREPL, ::Base.REPL.REPLBackendRef) at ./REPL.jl:945
 [15] run_repl(::Base.REPL.LineEditREPL, ::Base.##510#511) at ./REPL.jl:180
 [16] _start() at ./client.jl:413                                                                         
while loading no file, in expression starting on line 0       
```
on 11.2.  This seems to come from [here](https://github.com/JuliaData/DataFrames.jl/blob/master/src/abstractdataframe/show.jl#L38).

**Comment by nalimilan** [comment]
Care yo make a PR?

**Comment by nalimilan** [comment]
Though try on master first, as https://github.com/JuliaData/DataFrames.jl/pull/1320 should have fixed this.

**Comment by ExpandingMan** [comment]
There's another error in master right now.  In `DataFrame(; kwargs...)` the function `Compat.pairs` is returning elements of the form `n::Int=>(s::Symbol, x::Any)` when the constructor it's being provided to expects these to simply be `(s::Symbol, x::Any)`.  I assume this is for compatibility with 0.7 which passes keyword arguments as named tuples.  This simplest fix of this would be to replace line 119 in dataframe.jl with
```julia
DataFrame((k.second[1]=>v.second[2] for v in A)...)
```
but I'm assuming this would break 0.7.

**Comment by nalimilan** [comment]
@bkamins has just noted the same thing at https://github.com/JuliaData/DataFrames.jl/pull/1320.  I guess any solution which works on both 0.6 and 0.7 will do, given that everything is broken right now.

**Comment by ExpandingMan** [comment]
Ok... I'd really like to make a PR and actually be helpful, but I don't have 0.7 set up at all, so I'll have to leave it to someone else at least for now.

**Comment by ararslan** [comment]
The definition of `pairs` in Compat may be incorrect. We should check that first.

**Comment by ExpandingMan** [comment]
I suspect you are right.  The current output of `pairs` doesn't make a whole lot of sense.

**Comment by bkamins** [comment]
The problem with `pairs` and keyword arguments is that the API completely changed very much.

If in `Compat` we make `pairs` behave exactly like it behaves on 0.7 (it does not now) this will not fix the problem. Alo `Compat` currently does not define `keys` for `Tuple` and 0.7 does.

I do not know what best practice for such situations was  in the past but I guess that we will have to have a custom code in `DataFrames` anyway (but agreed that we first should resolve `Compat` issue).

**Comment by nalimilan** [comment]
See https://github.com/JuliaData/DataFrames.jl/pull/1329 about keyword arguments.

The `textwidth` deprecation warning on Julia 0.6 is a weird result of https://github.com/JuliaLang/Compat.jl/pull/432, I've commented there.

**Comment by nalimilan** [comment]
https://github.com/JuliaLang/Compat.jl/pull/437 fixes the `textwidth` warning on Julia 0.6.

**Comment by ExpandingMan** [comment]
This appears to be fixed on the most recent tagged versions of DataFrames (0.11.3) and Compat (0.42.0).
